iree/turbine/kernel/wave/utils/run_utils.py

```diff
@@ -50,24 +52,25 @@ def get_device_uuid(device_list: list[str], device_str: str) -> tuple[int, str]:
     return device_str
 
 
-def _invoke(vm_context, device, entry_function, inputs, outputs, dynamic_dims):
-    arg_list = rt.VmVariantList(len(inputs) + len(dynamic_dims))
+def _invoke(
+    vm_context, device, entry_function, inputs, outputs, scalar_args, dynamic_dims
+):
+    arg_list = rt.VmVariantList(len(inputs) + len(scalar_args) + len(dynamic_dims))
     ret_list = rt.VmVariantList(len(outputs))
 
-    for input in inputs:
+    for input in chain(inputs, scalar_args, dynamic_dims):
         if isinstance(input, torch.Tensor):
             input_cpu = input.cpu().contiguous()
             device_array = rt.asdevicearray(device, input_cpu)
             arg_list.push_ref(device_array._buffer_view)
+        elif isinstance(input, int):
+            arg_list.push_int(input)
+        elif isinstance(input, float):
+            warn("Currently, `push_float` is not working on the iree side")
```

Comment:
Why not throw an error here?

Reply:
To be able to write negative test and actually detect if it suddenly starts working.

Reply:
Got it, I see this is tested in test_scalar_codegen.